pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -122,7 +123,51 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 
 // admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	err := admitLimitRanges(ctx, repo, size)
+	if err != nil {
+		return err
+	}
+
+	err = admitQuotas(ctx, repo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
+			context.GetLogger(ctx).Warnf("Cannot list limitranges because of outdated cluster roles: %v", err)
```

Comment:
> I admit I am really confused by this as well. I would have expected what @soltysh said... @deads2k have you weighed in here? this seems weird.

I had not.  I disagree with @smarterclayton's comments in https://github.com/openshift/origin/issues/7797#issuecomment-199978462 .  If you use a new registry image with an old server, I think its ok to have that image fail (preferably preventing readiness so that deployment fails) until someone either disables the feature or fixes the roles.

@smarterclayton Why would I ever want to default to insecure?  I understand not wanting it to role out and be broken, so I can get behind failing a readiness check.